src/EventListener/SessionListener.php

```diff
@@ -45,12 +54,18 @@
     private $userIdentifierHeaders;
 
     /**
+     * @param ContainerInterface  $container             To allow to be extra lazy
      * @param BaseSessionListener $inner
      * @param string              $userHashHeader        Must be lower-cased
      * @param array               $userIdentifierHeaders Must be lower-cased
      */
-    public function __construct(BaseSessionListener $inner, string $userHashHeader, array $userIdentifierHeaders)
```

Comment:
You will have to adjust the extension as the argument indexes do not match anymore now.

Reply:
yep, wanted some feedback on the general approach before i adjust DI and the tests.